Python updates and relevant fixes for new HR tech tree port

temujin2k opened the pull request:
Hello, Leoreth!

I took some time to update the Python files in accordance with the new tech tree update. I have already finished adapting the spawn techs for every civ in all three scenarios to the new tech tree, but the lists I wrote there are based on my intuition and rough estimations of the equivalent spawn tech lists for the original tech tree. I also substituted many of the tech constants in the affected files. However I'm not so sure if I made the correct substitutions for the techs.

I've already tested a short game for each scenario with the new changes I made, and so far the game loads without errors, although minor Python exceptions may pop up along the way. I found that teching is still very slow here, so I think the DLL needs a lot of updating too. But I don't have any knowledge on C++ or editing the core DLL files so I won't be able to touch that.

_PS: I deleted, updated and pushed a new branch and then made a new pull request because I managed to screw up my commit history a few hours ago and the AppVeyor Bot checks that were too many messed up the pull request page. I'm very sorry about that!_

dguenms committed:
Initial tech integration

dguenms committed:
Own tech tree first iteration

dguenms committed:
Fix some tech references

dguenms committed:
Text and button adjustments

dguenms committed:
More tech buttons

dguenms committed:
Initial unit changes

dguenms committed:
Import unit changes from HR

dguenms committed:
Import unit art from HR

dguenms committed:
New units

dguenms committed:
Skirmisher and Horseman unit art

dguenms committed:
UU adjustments

temujin2k committed:
Updated Python constants in accordance with the HR tech tree port
    - Changed affected constants in related files
Updated religion founding techs (based on Google Sheets)
Updated spawn techs of all civs for the three scenarios
Updated unit references in Consts.py
Fixed no water working ability with Sailing

temujin2k committed:
Update CvRFCEventHandler.py tech references

dguenms commented:
Hey, thanks for the pull request. No need to worry about the builder bot, I don't think it's completely perfectly configured right now anyway.

Right now I'm not really in the state to be able to accept pull requests for the ongoing work though. Everything is still in a very fluid and experimental to be dealing with stuff like Python code, which is why I haven't touched it yet. And when I do, I really want to have a lot of conscious control over the changes I am making. Will return to your pull request for reference though when I get there.

temujin2k commented:
It's fine! I just update what needs to be updated whenever you push new commits. I guess changes you'll make to the tech tree will not need major Python revisions to this one, especially since the order of the techs in the XML made things much more convenient. Can't wait for the final tech tree and the relevant changes! I'll just keep this PR open just in case. :)

temujin2k committed:
Fixed Japanese DCN error

temujin2k committed:
Fixed HR tech advisor screen Python exception when updating research path

dguenms committed:
Improvement happiness and health

dguenms committed:
Improvement changes

dguenms committed:
More tech adjustments

dguenms committed:
Tech button changes

temujin2k committed:
Update constants to third iteration

dguenms committed:
New buildings from HR

dguenms committed:
Extra buildings

dguenms committed:
Building effects

- power commerce modifier
- power unhealth proportional to power consumed
- improvement unhealth/unhappiness

temujin2k committed:
Fixed Byzantium founding Islam on spawn
Minor Python fixes and updates

temujin2k committed:
Updated Python constants with new buildings
